src/otx/core/exporter/native.py

```diff
@@ -0,0 +1,142 @@
+        self.pad_value = pad_value
+        self.swap_rgb = swap_rgb
+        self.via_onnx = via_onnx
+        self.onnx_export_configuration = onnx_export_configuration if onnx_export_configuration is not None else {}
+
+    def _extend_model_metadata(self, metadata: dict[tuple[str, str], str]) -> dict[tuple[str, str], str]:
+        """Extends metadata coming from model with preprocessing-specific parameters.
+
+        Model's original metadata has priority over exporter's extra metadata
+
+        Args:
+            metadata (dict[tuple[str, str],str]): metadata to extend.
+
+        Returns:
+            dict[tuple[str, str],str]: updated metadata
+        """
+        mean_str = " ".join(map(str, self.mean))
+        std_str = " ".join(map(str, self.std))
+
+        extra_data = {
+            ("model_info", "mean_values"): mean_str.strip(),
+            ("model_info", "scale_values"): std_str.strip(),
+            ("model_info", "resize_type"): self.resize_mode,
+            ("model_info", "pad_value"): str(self.pad_value),
+            ("model_info", "reverse_input_channels"): str(self.swap_rgb),
+        }
+        extra_data.update(metadata)
+
+        return extra_data
+
+    def to_openvino(
+        self,
+        model: torch.nn.Module,
+        output_dir: Path,
+        base_model_name: str = "exported_model",
+        precision: OTXExportPrecisionType = OTXExportPrecisionType.FP32,
+        metadata: dict[tuple[str, str], str] | None = None,
+    ) -> Path:
+        """Export to OpenVINO Intermediate Representation format.
+
+        In this implementation the export is done only via standard OV/ONNX tools.
+        """
+        dummy_tensor = torch.rand(self.input_size).to(next(model.parameters()).device)
+
+        if self.via_onnx:
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                tmp_dir = Path(tmpdirname)
+
+                self.to_onnx(
+                    model,
+                    tmp_dir,
+                    base_model_name,
+                    OTXExportPrecisionType.FP32,
+                    None,
+                )
+                exported_model = openvino.convert_model(
+                    tmp_dir / (base_model_name + ".onnx"),
+                    input=(openvino.runtime.PartialShape(self.input_size),),
+                )
+        else:
+            exported_model = openvino.convert_model(
+                model,
+                example_input=dummy_tensor,
+                input=(openvino.runtime.PartialShape(self.input_size),),
+            )
+
+        # workaround for OVC's bug: single output doesn't have a name in OV model
+        if len(exported_model.outputs) == 1 and len(exported_model.outputs[0].get_names()) == 0:
+            exported_model.outputs[0].tensor.set_names({"output1"})
+
+        metadata = {} if metadata is None else self._extend_model_metadata(metadata)
+        exported_model = OTXNativeModelExporter._embed_openvino_ir_metadata(exported_model, metadata)
```

Comment:
```suggestion
        if metadata is not None:
            self._extend_model_metadata(metadata)
            exported_model = OTXNativeModelExporter._embed_openvino_ir_metadata(exported_model, metadata)
```
How about changing code like this to reduce function call?